crypto/init.c

```diff
@@ -729,3 +680,49 @@ int OPENSSL_atexit(void (*handler)(void))
     return 1;
 }
 
+/*
+ * Library  destructor (a.k.a. _fini()). Code here is derived
+ * from stuff found in providers/fips/self_test.c
+ *
+ * We do kind of  best effort cleanup. If given platform/compiler
+ * does not support module destructors then its up to application
+ * to explicitly call OPENSSL_cleanup();
+ */
+#if defined(__GNUC__) && !defined(_AIX)
+
+void call_cleanup(void);
+
+void __attribute__((destructor))
+call_cleanup(void)
```

Comment:
The function needs to be named `ossl_call_cleanup` or something like that - i.e. with the ossl prefix as it is non-static.